src/index.js

```diff
@@ -0,0 +1,156 @@
+  const cb = this.async();
+
+  const file = this.resourcePath;
+  const options = Object.assign({}, getOptions(this));
+
+  validateOptions('./src/options.json', options, 'CSS Loader');
+
+  if (options.sourceMap) {
+    if (map && typeof map !== 'string') {
+      map = JSON.stringify(map);
+    }
+  } else {
+    map = null;
+  }
+
+  // root: options.root
+  const processOptions = {
+    from: file,
+    to: file,
+    url: options.url,
+    import: options.import,
+    sourceMap: options.sourceMap
+  };
+
+  Promise.resolve()
+    .then(() => {
+      if (meta && meta.ast) return processor(ast, map, processOptions);
+      return processor(src, map, processOptions);
+    })
+    .then((result) => {
+      let css = JSON.stringify(result.css);
+
+      const imported = {};
+
+      const imports = result.imports
+        .filter((i) => {
+          if (!i.mediaQuery) {
+            if (imported[i.url]) return false;
+
+            imported[i.url] = true;
+          }
+
+          return true;
+        })
+        .map((i) => {
+          if(!isUrlRequest(i.url, root)) {
+            return `exports.push([module.id, ${JSON.stringify(`@import url(${i.url});`)}, ${JSON.stringify(`${i.mediaQuery}`)}]);`;
+          }
+        })
+        .join('\n');
+
+      function importMatcher (item) {
+        const match = result.importsRegExp.exec(item);
+        const idx = +match[1];
+
+        const url = result.imports[idx].url;
+
+        return `require('${stringifyRequest(this, url)}).locals' [${JSON.stringify(url.export)}]`;
+      }
+
+      css = css.replace(result.importsRegExpG, importMatcher.bind(this));
+
+      if (options.url !== false) {
+        css = css.replace(result.urlsRegExpG, (item) => {
+          const match = result.urlsRegExp.exec(item);
+
+          let idx = +match[1];
+
+          const urlItem = result.urls[idx];
+          const url = resolver(options.alias)(urlItem.url);
+
+          idx = url.indexOf('?#');
+
+          if (idx < 0) idx = url.indexOf('#');
+
+          if (idx > 0) {
+            // in cases like url('webfont.eot?#iefix')
+            const $url = url.substr(0, idx);
+
+            return `\" + require(${stringifyRequest(this, $url)}) + \"` + url.substr(idx);
+          }
+
+          return `\" + require(${stringifyRequest(this, url)}) + \"`;
+        });
+      }
+
+      let modules;
+
+      if (options.sourceMap && result.map) {
+        map = result.map;
+
+        if (map.sources) {
+          map.sourceRoot = '';
+          map.sources = map.sources
+            .map((source) => source.split('!').pop());
+        }
+
+        map.file = map.file.split('!').pop();
+        map = JSON.stringify(map);
+
+        modules = `exports.push([module.id, ${css}, '', ${map}]);`;
+      } else {
+        modules = `exports.push([module.id, ${css}, '']);`;
+      }
+
+      let locals;
+
+      if (meta && meta.locals) {
+        locals = `exports.locals = ${meta.locals};`;
```

Comment:
This is more of a placeholder yet :D :D I was in the process of trying out how to ES2015 modulify this. That's where we need to start afresh and if CSSM tsay in `css-loader` there will be no meta locals, since this was originally intended to coma from `postcss-loader`. Any other ideas how we can do it a highly welcome 😛 

```
let locals = []

Object.keys(meta.locals).forEach((key) => {
     locals.push `export const ${key} = ${meta.locals[key]}`
})

locals = locals.join('\n')
```